src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -256,7 +256,11 @@ const char *smbios_mainboard_serial_number(void)
 	// bridge and the first NIC is connected to the second PCIe bridge.
 	// Read secondary bus number from the PCIe bridge where the first NIC is
 	// connected.
-	dev = dev_find_slot(0, PCI_DEVFN(2, 2));
+	if (check_pciereverse())
+		dev = dev_find_slot(0, PCI_DEVFN(2, 3));
```

Comment:
@Sakoram I missed it in review. Note apu5 has only 2 NICs, if you try to look on PCIe 2.3 it will hit non-NIC device or notihing.